### PR TITLE
CONTRIBUTING: describe hlint and how it's used

### PR DESCRIPTION
**Pull request by tarleb**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -206,6 +206,32 @@ or Powerpoint to ensure that they weren't corrupted and that
...
 
+Code style
+----------
+
+Pandoc uses [hlint] to identify opportunities for code improvements,
+like redundant brackets, or pointing out unnecessary `Language`
+extensions. However, there sometimes are cases where there are good
+reasons to use code different from what hlint proposes. In these
+cases, the respective warning should be disabled in file
+`.hlint.yaml`.
```
Comment (alerque) [inline CONTRIBUTING.md:217]:
```suggestion
Pandoc uses [hlint] to identify opportunities for code improvements
like redundant brackets or unnecessary `Language`
extensions. However, sometimes there are cases where there are good
reasons to use code different from what hlint proposes. In these
cases, the respective warning should be disabled in the file
`.hlint.yaml`.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -206,6 +206,32 @@ or Powerpoint to ensure that they weren't corrupted and that
...
+
+There should be no errors when running `hlint .`; this is checked by
+the continuous integration (CI) setup. It is recommended that
+contributors check their code with a local hlint installation, but
+relying on the CI is fine, too.
+
+A good way to ensure no new hints are triggered is to use a git
+[pre-commit hook], which runs hlint on all updated Haskell files
```
Comment (alerque) [inline CONTRIBUTING.md:225]:
```suggestion
A good way to ensure no new warnings are introduced is to use a Git
[pre-commit hook] which runs hlint on all updated Haskell files
```

**Review by alerque — CHANGES_REQUESTED** [review]
Relatively minor English copy-editing.

File: CONTRIBUTING.md (hunk 2 above)
Comment (tarleb) [reply CONTRIBUTING.md]: Thanks!

File: CONTRIBUTING.md (hunk 1 above)
Comment (tarleb) [reply CONTRIBUTING.md]: Thanks, updated!

**Review by alerque — APPROVED** [review]

**Pull request merged**